wallet/types/src/keys.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::key::extended::{ExtendedPrivateKey, ExtendedPublicKey};
+use serialization::{Decode, Encode};
+use storage::HasPrefix;
+
+/// The key id prefix is used for searching specific kinds of keys
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub enum KeyIdPrefix {
+    #[codec(index = 0)]
+    DeterministicRoot,
+}
```

Comment:
I'd probably call this `KeyType` or `KeyClass`. The fact that it's encoding is a prefix of encoding of another type seems like an internal detail which is worth mentioning in comment but not important enough to be a part of the name imho.

Reply:
I removed this